hack/windows/archive-proxy/main.go

```diff
@@ -0,0 +1,95 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/elastic/elastic-agent/pkg/core/process"
+)
+
+// CommitSHA is set by the linker at build time
+var CommitSHA string
+
+func main() {
+	if CommitSHA == "" {
+		// this should never happen
+		log.Fatal("No commit SHA provided")
+	}
+
+	exePath, err := os.Executable()
+	if err != nil {
+		log.Fatalf("Error getting executable path: %v\n", err)
+	}
+
+	exeAbsPath, err := filepath.Abs(exePath)
+	if err != nil {
+		log.Fatalf("Error getting executable absolute path: %v\n", err)
+	}
+
+	// Fabricate the elastic-agent.exe path that reside inside the data/elastic-agent-{commit-short-sha} directory
+	exeTopPath := filepath.Dir(exeAbsPath)
+	nestedAgentBinaryPath := filepath.Join(exeTopPath, "data", fmt.Sprintf("elastic-agent-%s", CommitSHA), "elastic-agent.exe")
+	if _, err := os.Stat(nestedAgentBinaryPath); err != nil {
+		log.Fatalf("Unable to stat inner agent binary at %q: %v\n", nestedAgentBinaryPath, err)
```

Comment:
```suggestion
		log.Fatalf("Unable to execute elastic-agent.exe at %q: %v\n", nestedAgentBinaryPath, err)
```

stat has no meaning to Windows users who have never used Linux :)

Reply:
valid point, reworded according to proposal here 316fc118304a26ff4e267a6dca316fd2890bd888, please feel free to unresolve if you spot anything else with it 🙂